Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/file-uploader/FileUploader.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ChangeEvent, useCallback, useEffect, useRef, useState } from 'react';
+import isNil from 'lodash/isNil';
+
+import { useSubmissions } from '../../../hooks/submissions/use-submissions';
+import Button, { ButtonSize, ButtonType } from '../buttons/Button';
+
+import styles from './FileUploader.module.scss';
+
+interface IFileUploaderProps {
+    file?: File | null;
+    problemId?: number;
+}
+
+const FileUploader = ({ file, problemId }: IFileUploaderProps) => {
+    const hiddenFileInput = useRef<HTMLInputElement | null>(null);
+    const { actions: { updateSubmissionCode } } = useSubmissions();
+    const [ internalFile, setInternalFile ] = useState<File | null>(null);
+    const [ internalProblemId, setInternalProblemId ] = useState<number | null>(null);
+    const handleClick = () => {
+        hiddenFileInput.current?.click();
+    };
+
+    useEffect(
+        () => {
+            if (problemId !== internalProblemId && isNil(file)) {
+                setInternalFile(null);
+                return;
+            }
+
+            if (!isNil(file) && !isNil(problemId)) {
+                setInternalFile(file);
+                setInternalProblemId(problemId);
+            }
+        },
+        [ file, internalFile, internalProblemId, problemId ],
+    );
+
+    const handleChange = useCallback(
+        (event: ChangeEvent<HTMLInputElement>) => {
+            const { target: { files: eventTarget } } = event;
+            if (!eventTarget || isNil(problemId)) {
+                return;
+            }
+
+            updateSubmissionCode(eventTarget[0]);
+            setInternalFile(eventTarget[0]);
+            setInternalProblemId(problemId);
+        },
+        [ updateSubmissionCode, problemId ],
+    );
+
+    return (
+        <>
+            <div className={styles.fileUploadContainer}>
+                <div>
+                    <Button
+                      onClick={handleClick}
+                      type={ButtonType.submit}
+                      size={ButtonSize.medium}
+                    >
+                        Click to select
+                    </Button>
+                </div>
+                <div className={styles.fileName}>
+                    {isNil(internalFile)
+                        ? ''
+                        : internalFile.name}
+                </div>
+            </div>
+            <input
```

Comment:
There is already an input type file in the source code of the app. Check if you can use whats there.

Reply:
Unfortunately I need a component with a different logic and I cannot use it.